frontend/src/hooks/useFeatureFlags.ts

```diff
@@ -1,48 +1,70 @@
+"use client";
+
 import Cookies from "js-cookie";
-import { FeatureFlagsManager } from "src/services/FeatureFlagManager";
+import { isBoolean } from "lodash";
+import {
+  defaultFeatureFlags,
+  FeatureFlags,
+} from "src/constants/defaultFeatureFlags";
+import {
+  FEATURE_FLAGS_KEY,
+  getCookieExpiration,
+} from "src/services/featureFlags/featureFlagHelpers";
 
-import { useEffect, useState } from "react";
+import { useCallback, useEffect, useState } from "react";
 
 /**
- * React hook for reading and managing feature flags in client-side code.
- *
- * ```
- * function MyComponent() {
- *   const {
- *     featureFlagsManager,  // An instance of FeatureFlagsManager
- *     mounted,  // Useful for hydration
- *     setFeatureFlag,  // Proxy for featureFlagsManager.setFeatureFlagCookie that handles updating state
- *   } = useFeatureFlags()
- *
- *   if (featureFlagsManager.isFeatureEnabled("someFeatureFlag")) {
- *     // Do something
- *   }
- *
- *   if (!mounted) {
- *     // To allow hydration
- *     return null
- *   }
+ * Allows client components to access feature flags by
+ *  - setting the cookie
+ *  - reading the cookie
  *
- *   return (
- *     ...
- *   )
- * }
- * ```
  */
-export function useFeatureFlags() {
-  const [featureFlagsManager, setFeatureFlagsManager] = useState(
-    new FeatureFlagsManager(Cookies),
-  );
-  const [mounted, setMounted] = useState(false);
+export function useFeatureFlags(): {
+  setFeatureFlag: (flagName: string, value: boolean) => void;
+  checkFeatureFlag: (flagName: string) => boolean;
+  featureFlags: FeatureFlags;
+} {
+  const [featureFlags, setFeatureFlags] =
+    useState<FeatureFlags>(defaultFeatureFlags);
 
+  // a workaround, as setting this in default state value results in hydration error
+  // on feature flag admin page. Does it cause a blip on other pages?
```

Comment:
this seems fine but it's a workaround just for the admin page so it may not be necessary